### ISSUE TITLE
Dynamic import does not work after 2.0.7

### ISSUE THREAD

**Issue by jackysee** [webpack-contrib/karma-webpack#291]
**I'm submitting a bug report**

**Webpack version:**
3.10.0

**Webpack Karma version**:
2.0.7

**Karma version**:
1.7.1

**Please tell us about your environment:**
Windows 10

**Browser:** Chrome

**Current behavior:**
Having the webpack do the code splitting at dynamic import statement e.g. `import('./file.js')`, karma serve the path as system file path and chrome report that it cannot load the resource:
```
Not allowed to load local resource: file:///C:/Users/jackysee/AppData/Local/Temp/_karma_webpack_/0.bundle.js
```

**Expected/desired behavior:**
It should  load the resource at e.g. `http://localhost:9867/_karma_webpack/0.bundle.js`, just as it did in version karma-webpack 2.0.6.

* **demo**

https://github.com/jackysee/karam-webpack-test

**Comment by ldrick** [comment]
I've got the same problem on a Ubuntu 17.10 with ChromeHeadless and a Windows with ChromeHeadless.

Had to go back to 2.0.7, because 2.0.8 didn't work either.
Seems this [0616dda](https://github.com/webpack-contrib/karma-webpack/commit/0616dda) somehow broke dynamic imports.

**Comment by rogeriochaves** [comment]
+1, same problem here, on phantomjs macosx and phantomjs on jenkins (some linux distro)

**Comment by rodmax** [comment]
i hope i fixed this in #292 
Please check if this is a valid solution.

**Comment by rodmax** [comment]
@jackysee i tested https://github.com/jackysee/karam-webpack-test with #293 pull-request
It fixed this bug on my OS(Ubuntu). and i believe on MACs too

But i can't check it on Windows.
Could you(or somebody else:) ) verify it on Windows, because I have doubts
that the bug may remain on Windows due to the  temp folder is a more complicated string than in Unix.

below steps may help you to quick setup env
```
git clone git@github.com:rodmax/karma-webpack.git
cd karma-webpack
git checkout fix-url-regex-for-custom-file-handler
npm i 
npm i webpack@3.6.0   # you should install peer dependency for using "npm link"
npm link
npm run build  # after it you should see "js" files in ./lib dir

cd path/to/karam-webpack-test
npm link karma-webpack  # mount my pull-requested version
npm run tests
```

**Comment by jackysee** [comment]
Tested, doesn't work on windows.
Console shows `Not allowed to load local resource: file:///C:/Users/jackys/AppData/Local/Temp/_karma_webpack_/0.bundle.js`

![image](https://user-images.githubusercontent.com/30516/34281941-110bdcf2-e6fd-11e7-8e01-55e012bc7360.png)

**Comment by rodmax** [comment]
@jackysee 
thanks for quick reply, as i expected, my fix working only for unix systems, 

For Windows needed more complex fix due to `os.tmpdir()` on Windows returns absolute path (`C:/....` or something else) and browser attempts to load local file instead of request to `localhost:XXX`, and this request not accepted karma server at all

**Comment by sjd78** [comment]
If it helps, the karma-webpack generated chunk asset files don't actually live on disk.  They live in memory via webpack-dev-middleware.  When I do something like this:

```javascript
require.ensure([], (require) => {
   require('module1')
   require('module2')
   require('module3')
}, 'named-chunk')
```

the karma-webpack setup will generate `0.bundle.js` as the asset for chunk `named-chunk` with those modules embedded to be lazy loaded when/if needed.  Then my test code won't ever be able to load `0.bundle.js` because it's looking for it on disk, and it's not there.  But it is accessible on the webpack-dev-server at `http://0.0.0.0:9876/_karma_webpack_/0.bundle.js`.

It does work on 2.0.6.

**Comment by johnnyreilly** [comment]
Thanks so much for reporting this!  I've switch back to 2.0.6 for now - my tests pass once more 👍 (Windows again)

**Comment by johman10** [comment]
I can confirm that #293 fixes the issue on Mac in more instances. Would be great if we can get this fixed soon-ish so that we can update.

I can also confirm that downgrading to 2.0.6 actually works to fix this as well, it's not perfect but it's something.

**Comment by michael-ciniawsky** [comment]
Fix released in [`v2.0.10`](https://github.com/webpack-contrib/karma-webpack/releases/tag/v2.0.10)  🎉  Please try it out. If still regressions please open a new issue :)
